src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,130 @@
+                    + product.getName(), e);
+        }
+        return product;
+    }
+
+    @Override
+    public Optional<Product> get(Long id) {
+        String query = "SELECT * FROM products WHERE product_id = ? AND deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                Product product = getProductFromSet(resultSet);
+                return Optional.of(product);
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create product with id "
+                    + id, e);
+        }
+    }
+
+    @Override
+    public List<Product> getAll() {
+        String query = "SELECT * FROM products WHERE deleted = false";
+        List<Product> productList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            ResultSet resultSet = preparedStatement.executeQuery(query);
+            while (resultSet.next()) {
+                Product product = getProductFromSet(resultSet);
+                productList.add(product);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Incorrect getAll query", e);
+        }
+        return productList;
+    }
+
+    @Override
+    public Product update(Product product) {
+        String query = "UPDATE products SET product_name = ?, product_price = ? "
+                + "WHERE product_id = ? AND deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setString(1, product.getName());
+            preparedStatement.setDouble(2, product.getPrice());
+            preparedStatement.setLong(3, product.getId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Incorrect update query", e);
+        }
+        return product;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "UPDATE products SET deleted = true WHERE product_id = ?"
+                + " AND deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Incorrect deleteById query", e);
```

Comment:
```suggestion
            throw new DataProcessingException("Can't delete product by id " + id, e);
```
use same funnel with other exception messages